File: src/main/java/nl/rabobank/assessment/persistence/repository/PetRepository.java
package nl.rabobank.assessment.persistence.repository;

import nl.rabobank.assessment.persistence.entity.Pet;
import org.springframework.data.repository.reactive.ReactiveCrudRepository;
import reactor.core.publisher.Flux;

import java.util.UUID;

public interface PetRepository extends ReactiveCrudRepository<Pet, UUID> {

	Flux<Pet> findPetsByPersonId(UUID personId);
}

File: src/main/kotlin/nl/rabobank/assessment/persistence/repository/PetRepository.kt
package nl.rabobank.assessment.persistence.repository

import nl.rabobank.assessment.persistence.entity.Pet
import org.springframework.data.repository.reactive.ReactiveCrudRepository
import reactor.core.publisher.Flux
import java.util.*

interface PetRepository : ReactiveCrudRepository<Pet, UUID> {

    fun findPetsByPersonId(personId: UUID): Flux<Pet>
}